hca/util/__init__.py

```diff
@@ -233,6 +233,7 @@ def store_refs(d):
     def swagger_spec(self):
         if not self._swagger_spec:
             swagger_url = self.config[self.__class__.__name__].swagger_url
+            temp_swagger_filename = os.path.join(self.config.user_config_dir, "temp_swagger.json")
```

Comment:
I would use `with tempfile.NamedTemporaryFile`

Reply:
Cool. Thanks!

Reply:
Oof you should then use it with the option to use the user config dir as the tempdir. Otherwise the rename is not atomic either (since it could actually be a copy).